test/test_shape_ops.py

```diff
@@ -338,86 +338,137 @@ def test_clamp_raises_arg_errors(self, device):
-
-        # test for shape
-        data = torch.randn(2, 3, 4, device=device)
-        size = [2, 3, 4]
-        test_dims = []
-        for i in range(1, 3):
-            test_dims += combinations(range(len(size)), i)
-
-        for ds in test_dims:
-            self.assertEqual(size, list(data.flip(ds).size()))
-
-        # test rectangular case
-        data = torch.tensor([1, 2, 3, 4, 5, 6], device=device).view(2, 3)
-        flip0_result = torch.tensor([[4, 5, 6], [1, 2, 3]], device=device)
-        flip1_result = torch.tensor([[3, 2, 1], [6, 5, 4]], device=device)
-
-        self.assertEqual(flip0_result, data.flip(0))
-        self.assertEqual(flip1_result, data.flip(1))
-
-        # test empty tensor, should just return an empty tensor of the same shape
-        data = torch.tensor((), device=device)
-        self.assertEqual(data, data.flip(0))
-
-        # test bool tensor
-        a = torch.tensor([False, True], device=device)
-        self.assertEqual(a.flip(0), torch.tensor([True, False]))
-
-        # case: dims=()
-        a = torch.randn(3, 2, 1, device=device)
-        self.assertEqual(a.flip(dims=()), a)
 
     def _rand_shape(self, dim, min_size, max_size):
-        shape = []
-        for i in range(dim):
-            shape.append(random.randint(min_size, max_size))
-        return tuple(shape)
-
-    @dtypes(torch.cfloat, torch.cdouble)
-    def test_complex_flip(self, device, dtype):
-        rand_dim = random.randint(3, 4)
-        shape = self._rand_shape(rand_dim, 5, 10)
-
-        # Axis to sample for given shape.
-        for i in range(1, rand_dim):
-            # Check all combinations of `i` axis.
-            for flip_dim in combinations(range(rand_dim), i):
-                data = torch.randn(*shape, device=device, dtype=dtype)
-                torch_fn = partial(torch.flip, dims=flip_dim)
-                np_fn = partial(np.flip, axis=flip_dim)
-                self.compare_with_numpy(torch_fn, np_fn, data)
+        return tuple(torch.randint(min_size, max_size + 1, (dim,)))
+
+    # Do not include bool as address sanitiser complains
+    @dtypes(*torch.testing.get_all_dtypes(include_bool=False))
+    def test_flip_numpy(self, device, dtype):
+        make_arg = partial(make_tensor, dtype=dtype, device=device)
+
+        for ndim in [3, 4]:
+            shape = self._rand_shape(ndim, 5, 10)
+            data = make_arg(shape)
+
+            # Axis to sample for given shape.
+            for i in range(1, ndim + 1):
+                # Check all combinations of `i` axis.
+                for flip_dim in combinations(range(ndim), i):
+                    torch_fn = partial(torch.flip, dims=flip_dim)
+                    np_fn = partial(np.flip, axis=flip_dim)
+                    self.compare_with_numpy(torch_fn, np_fn, data)
+
+    @onlyCUDA  # CPU is too slow
+    @largeTensorTest('30GB')
+    @dtypes(torch.int64, torch.double, torch.cdouble)
+    def test_flip_large_tensor(self, device, dtype):
```

Comment:
Is 30GB GPU memory enough?
sizeof(cdouble) * 2**32 = 64GiB.
I suggest testing only on `torch.uint8`, and initialize the tensor with 
```python
torch.empty(2**32 + 1, dtype=dtype).random_()
```
instead of just empty tensor.

Reply:
Or, if you want to test `torch.double` you can use fewer elements. Splitting happens based on the extent in bytes, not in elements, you can see that even now for any unary op - any input more than 2GB will launch 2 kernels.